src/cljc/contentql/core.cljc

```diff
@@ -189,24 +186,21 @@
                :root false)))
 
 (defn ^:private reducer
-  "This reducer is te core of the recursive transformation. It uses the linked entries
+  "This reducer is the core of the recursive transformation. It uses the linked entries
   and linked assets provided as part of the partial injection to create the map tree."
   [{:keys [linked-entries linked-assets] :as options} accum k v]
   (let [new-key (->kebab-case-keyword (name k))]
     (assoc accum new-key
                  (cond
-                   ; There are scenarios in which you can have a map that is not an Asset
-                   ; (such as a single reference field).
-                   ; These two functions (reduce-map-asset? & reduce-map-not-asset?)
-                   ; will differentiate between the two and apply the right transformations.
+                   ;; TODO Rich text does not return the right data
                    (reduce-map-asset? v) (transform-image v linked-assets)
 
-                   (reduce-map-not-asset? v) (let [vector-v (vector v)]
-                                               (transform-collection vector-v options linked-entries))
+                   (reduce-map-entry? v) (first (transform-collection (vector v) options linked-entries))
 
-                   (reduce-collection? v) (transform-collection v options linked-entries)
+                   (reduce-collection-entry? v) (transform-collection v options linked-entries)
+
+                   (reduce-collection-asset? v) (mapv #(transform-image % linked-assets) v)
 
-                   (reduce-nested-string? v) (first v)
```

Comment:
I tried all the different field types in Contentful to see where this would apply, and I couldn't see where this reducer helped (it actually caused the issue with the list of strings instead)